pgmpy/inference/base.py

```diff
@@ -80,3 +80,40 @@ def __init__(self, model):
             for factor in model.get_factors():
                 for var in factor.variables:
                     self.factors[var].append(factor)
+
+
+class ApproxInference:
+    """
+    Base class for the approximate inference algorithm.
+    Parameters
+    ----------
+    model: pgmpy.models.MarkovModel
+    We need a pairwise MarkovModel for Approx. inference.
+    """
+    def __init__(self, model):
+
+        # List of the variables in the present pairwise Markov Model
+        self.variables = []
+
+        # List of the cardinalities of the above variables in that order
+        self.cardinality = []
```

Comment:
You don't need to compute the cardinality of all the variables again. I think all the models have ```check_model()``` method, which computes the cardinality of the variables. Have a look at ```MarkovModel.check_model()``` method.